### PR TITLE
Simplify some control flow code

### PR DESCRIPTION
**Pull request by Deycoesr**
I found some control flow code can be simplify

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@Deycoesr Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=24907)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=24907) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@Deycoesr Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=24907)!

File: spring-web/src/main/java/org/springframework/web/bind/support/WebRequestDataBinder.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ public void bind(WebRequest request) {
 	 */
 	private boolean isMultipartRequest(WebRequest request) {
 		String contentType = request.getHeader("Content-Type");
-		return (contentType != null && StringUtils.startsWithIgnoreCase(contentType, "multipart"));
+		return StringUtils.startsWithIgnoreCase(contentType, "multipart");
```
Comment (sbrannen) [inline spring-web/src/main/java/org/springframework/web/bind/support/WebRequestDataBinder.java:133]: Good catch

File: spring-web/src/main/java/org/springframework/web/filter/OncePerRequestFilter.java [hunk 2]
```diff
@@ -129,10 +129,7 @@ private boolean skipDispatch(HttpServletRequest request) {
 		if (isAsyncDispatch(request) && shouldNotFilterAsyncDispatch()) {
 			return true;
 		}
-		if (request.getAttribute(WebUtils.ERROR_REQUEST_URI_ATTRIBUTE) != null && shouldNotFilterErrorDispatch()) {
-			return true;
-		}
-		return false;
+		return request.getAttribute(WebUtils.ERROR_REQUEST_URI_ATTRIBUTE) != null && shouldNotFilterErrorDispatch();
```
Comment (sbrannen) [inline spring-web/src/main/java/org/springframework/web/filter/OncePerRequestFilter.java:132]: Although the proposed changed is technically correct, please revert this for the sake of readability.

File: spring-web/src/main/java/org/springframework/web/jsf/el/SpringBeanFacesELResolver.java [hunk 3]
```diff
@@ -122,9 +122,7 @@ public boolean isReadOnly(ELContext elContext, @Nullable Object base, Object pro
 		if (base == null) {
 			String beanName = property.toString();
 			WebApplicationContext wac = getWebApplicationContext(elContext);
-			if (wac.containsBean(beanName)) {
-				return true;
-			}
+			return wac.containsBean(beanName);
```
Comment (sbrannen) [inline spring-web/src/main/java/org/springframework/web/jsf/el/SpringBeanFacesELResolver.java:125]: Although the proposed changed is technically correct, please revert this for the sake of readability.

File: spring-web/src/main/java/org/springframework/web/multipart/support/MultipartResolutionDelegate.java [hunk 4]
```diff
@@ -156,10 +156,7 @@ private static boolean isPartArray(MethodParameter methodParam) {
 	private static Class<?> getCollectionParameterType(MethodParameter methodParam) {
 		Class<?> paramType = methodParam.getNestedParameterType();
 		if (Collection.class == paramType || List.class.isAssignableFrom(paramType)){
-			Class<?> valueType = ResolvableType.forMethodParameter(methodParam).asCollection().resolveGeneric();
-			if (valueType != null) {
-				return valueType;
-			}
+			return ResolvableType.forMethodParameter(methodParam).asCollection().resolveGeneric();
```
Comment (sbrannen) [inline spring-web/src/main/java/org/springframework/web/multipart/support/MultipartResolutionDelegate.java:159]: Although the proposed changed is technically correct, please revert this for the sake of readability.

**Review by sbrannen — CHANGES_REQUESTED** [review]
Thanks for the PR!

I requested that you revert a few of the changes for the sake of readability.

Once you make those changes, I'll be glad to merge this PR.

**Comment by sbrannen** [comment]
@Deycoesr, did you intentionally close this PR?

**Comment by Deycoesr** [comment]
Not, I used `git push origin HEAD --force`.

**Comment by Deycoesr** [comment]
I'm sorry, i don't know there will be such consequences. I created a new PR. #24925

**Pull request closed without merging**